### PR TITLE
FEATURE: Add include_subcategories report filter

### PR DESCRIPTION
**Pull request by udan11**
Adds a new filter to reports with a Category filter:

### CODE REVIEW

File: app/assets/javascripts/admin/components/report-filters/include-subcategories.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  },
+
+  @discourseComputed()
+  disabled() {
+    return this.model.available_filters.any(
+      filter => filter.id === "category" && !filter.default
+    );
+  },
+
+  actions: {
```
Comment (jjaffeux) [inline app/assets/javascripts/admin/components/report-filters/include-subcategories.js:18]: use @action please

File: app/assets/javascripts/admin/components/report-filters/include-subcategories.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import FilterComponent from "admin/components/report-filters/filter";
+import discourseComputed from "discourse-common/utils/decorators";
+
+export default FilterComponent.extend({
+  checked: false,
+
+  didReceiveAttrs() {
+    this.set("checked", this.filter.default === "true");
```
Comment (jjaffeux) [inline app/assets/javascripts/admin/components/report-filters/include-subcategories.js:8]: needs this._super(...arguments)

File: app/assets/javascripts/admin/components/report-filters/include-subcategories.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  },
+
+  @discourseComputed()
```
Comment (jjaffeux) [inline app/assets/javascripts/admin/components/report-filters/include-subcategories.js:11]: would prefer using direct ember computed here, you don't use anything discourseComputed macro brings

**Comment by jjaffeux** [comment]
LGTM :+1: happy to filters having a good use case

**Comment by udan11** [comment]
I am closing this in favour of #9444. @jjaffeux, I will be implementing your feedback in the other pull request. Thank you for the review!

**Pull request closed without merging**